### ISSUE TITLE
app.use(<path>, <Router>) is Incompatible with Express

### ISSUE THREAD

**Issue by hpandelo** [tinyhttp/tinyhttp#400]
The Router() should work by appending on app via `app.use` like we do in ExpressJS

app.ts file:
```ts
import { App } from '@tinyhttp/app'
import { logger } from '@tinyhttp/logger'
import { pushMiddleware } from '@tinyhttp/router'
import { MainRouter } from './modules/main/index.ts'

export const app = new App()

app.use(logger())

// pushMiddleware(MainRouter.middleware) /* Not Working */
// app.use(MainRouter) /* Not Working */
// app.use('/home', MainRouter) /* Not Working */
// app.use(MainRouter.middleware) /* Also Not Working */

app.middleware.push(...MainRouter.middleware) /* Working adding to root path */

app
  .route('/home')
  .middleware.push(...MainRouter.middleware) /* Working adding to home path */
```

main router file:
```ts
import { Request, Response } from '@tinyhttp/app'
import { Router } from '@tinyhttp/router'

const router = new Router()

router.get('/', (_req: Request, res: Response) => {
  res.status(200).json({ message: 'Welcome' })
})

export const MainRouter = router

```

**Comment by hpandelo** [comment]
Also good to mention:

```ts
import { App, Request, Response } from '@tinyhttp/app'

const router = new App().route('/home') /* Won't register the /home */

router.get('/', (_req: Request, res: Response) => {
  res.status(200).json({ message: 'Welcome' })
})

export const MainRouter = router
```

To have a way to set the base/root path together into the Router file will be awesome

**Comment by aarontravass** [comment]
Hi @hpandelo thanks for reporting these errors. I do have 2 things to point out

> // pushMiddleware(MainRouter.middleware) /* Not Working */

`pushMiddleware` is an internal method and is generally not supposed to be used by the user. Moreover, the syntax you have used it incorrect as well. See below for it's usage internally.

https://github.com/tinyhttp/tinyhttp/blob/54d95f08fb6a2f8f5e7c017e5017be96bbeff6a6/packages/router/src/index.ts#L212-L218

> // app.use(MainRouter.middleware) /* Also Not Working */

`MainRouter.middleware` is an array and `app.use` uses the spread operator. Hence calling `app.use(MainRouter.middleware)` results in `app.use` receiving `[[//...middleware]]` which is incorrect. Nevertheless, using the spread operator throws an error.

**Comment by aarontravass** [comment]
One way to use `app.use(MainRouter.middleware)` would be to extract the handler and then pass. Like this
`app.use(MainRouter.middleware.map(m => m.handler))`

**Comment by hpandelo** [comment]
There are some ways to make it work, but won't be compatible with the usage in Express

**Comment by aarontravass** [comment]
> but won't be compatible with the usage in Express

Yes. There's also an existing issues but that is related to types #387

**Comment by hpandelo** [comment]
Another point, by using this way:
`app.route('/home').middleware.push(...MainRouter.middleware)`

Middlewares and settings wont work
```ts
export const app = new App({
  noMatchHandler: ...
  onError: ...
  settings: {
    networkExtensions: true,
    xPoweredBy: 'NeoX',
  },
})
```

`xPoweredBy` for example with a custom string or with `false` (also with `app.use(helmet()`) only will be affecting the routes from `app.ts` file, all the others "pushed" from `Router()` won't work (still returning the `x-powered-by: tinyhttp`)

**Comment by aarontravass** [comment]
> Middlewares and settings wont work

Yes, that is due to the fact that `route` returns a new instance.

https://github.com/tinyhttp/tinyhttp/blob/54d95f08fb6a2f8f5e7c017e5017be96bbeff6a6/packages/app/src/app.ts#L264-L270

**Comment by talentlessguy** [comment]
tinyhttp author here

a few things:

1. `@tinyhttp/router` is an internal package and is a barebones router without most of the tinyhttp's logic. basically it's just a fancy thing to manage middleware arrays. you can't use both `app` and `router`. Just use the `app` package.

2. `app.use(<path>, <app>)` is working as expected and passes in tests, see example https://github.com/tinyhttp/tinyhttp/blob/master/examples/subapps

basically the correct way is this:

```js
const app = new App()
const subapp = new App()
subapp.get('/', (req, res) => res.json({ hello: 'world' }))
app.use(subapp)
```

4. subapps created via `app.route` is a bug and should be filed as a separate issue (PR welcome)

if there's anything left unclear, pls lmk
